narwhals/dataframe.py

```diff
@@ -112,6 +119,9 @@ def drop_nulls(self) -> Self:
 
     @property
     def columns(self) -> list[str]:
+        if self._is_polars and self._backend_version >= (1, 0, 0):
+            return self._compliant_frame.collect_schema().names()  # type: ignore[no-any-return]
```

Comment:
not sure about this - like this we'd be hiding that `.columns` isn't a free operation for lazyframes

shouldn't we also define a `Schema` object and return that from `collect_schema`, and let Polars issue its warning for `.columns` / `.schema`?

Reply:
Oh I see your point! Of course that makes total sense - I will give it a try in the next few days 👌